src/polyswarm_api/resources.py

```diff
@@ -34,13 +34,37 @@ class Engine(core.BaseJsonResource):
 
     def __init__(self, content, api=None):
         super(Engine, self).__init__(content=content, api=api)
-        self.address = content['address'].lower()
-        self.name = content.get('name')
+        self.id = str(content['id'])
+        self.name = content['name']
+
+        if 'address' in content and isinstance(content['address'], str):
+            self.address = content['address'].lower()
+        else:
+            self.address = None
+
+        self.engine_type = content.get('engineType')
+        self.verified = content.get('status') == 'verified'
+
+        # These fields can be `null`; don't replace w/ default value in `get()`
+        self.artifact_types = tuple(content.get('artifactTypes') or ())
+        self.tags = tuple(content.get('tags') or ())
+        self.communities = tuple(content.get('communities') or ())
 
     @classmethod
     def _list_headers(cls, api):
         return {'Authorization': None}
 
+    def __hash__(self):
+        return hash(self.id)
+
+    def __eq__(self, other):
+        if isinstance(other, Engine):
+            return self.id == other.id
+        return False
+
+    @property
```

Comment:
Please don't create a property for this, this is not part of the json response is not a materialized attribute. Either use it as a method or materialize it in the `__init__` (I dislike this last option since we would be calculating stuff even if the user does not read it). Save the `@property` for caching attributes with the same name.